pylint/checkers/typecheck.py

```diff
@@ -124,32 +123,8 @@ def _is_owner_ignored(
     matches any name from the *ignored_classes* or if its qualified
     name can be found in *ignored_classes*.
     """
-    ignored_modules = set(ignored_modules)
-    module_name = owner.root().name
-    module_qname = owner.root().qname()
-
-    for ignore in ignored_modules:
-        # Try to match the module name / fully qualified name directly
-        if module_qname in ignored_modules or module_name in ignored_modules:
-            return True
-
-        # Try to see if the ignores pattern match against the module name.
-        if fnmatch.fnmatch(module_qname, ignore):
-            return True
-
-        # Otherwise, we might have a root module name being ignored,
-        # and the qualified owner has more levels of depth.
-        parts = deque(module_name.split("."))
-        current_module = ""
-
-        while parts:
-            part = parts.popleft()
-            if not current_module:
-                current_module = part
-            else:
-                current_module += f".{part}"
-            if current_module in ignored_modules:
-                return True
+    if is_module_ignored(owner.root(), ignored_modules):
```

Comment:
I like the new function, but could it be a refactor in another MR please ? So if we need to revert later we don't also revert the refactor and for easier review now :)

Reply:
Not sure what you mean, do you want me to duplicate the code instead?

Reply:
Ah I get it now, yes I'll do that

Reply:
Done: #7604 (contains the first commit of this PR)

Reply:
Thank you, would you mind rebasing this one so GitHub realize that one of the commit is now on main, please ?